### PR TITLE
Upgrade to Bootstrap 3

### PR DESCRIPTION
**Pull request by justinvdm**
Diamondash currently uses Bootstrap 3. Although it would probably take less work to downgrade diamondash to Bootstrap 2, we should probably be upgrading to Bootstrap 3 before starting on the new UI changes we have planned.

### CODE REVIEW

**Comment by justinvdm** [comment]
Ready for review. Although there were quite a few changes, the addition and deletion statistics are probably skewed by the vendor file changes in `go/base/static/vendor` after upgrading Bootstrap and Bootbox. Wondering if there is something we could do (cherry picking or something?) to exclude the vendor file changes and make reviewing easier?

**Comment by hodgestar** [comment]
I think it's easy enough to just scroll past the bits in `vendor` for now.

**Comment by hodgestar** [comment]
We should fix the failing Travis tests (some seem related).

File: go/apps/jsbox/view_definition.py [hunk 1]
```diff
@@ -23,11 +23,23 @@ def get(self, request, conversation):
...
+    def get(self, request, conversation):
+        edit_forms = dict(self.make_forms_dict(conversation))
+
+        return self.render_to_response({
+            'conversation': conversation,
+            'jsbox_form': edit_forms['jsbox'],
+            'jsbox_app_config_forms': edit_forms['jsbox_app_config'],
+            'edit_forms_media': self.sum_media(edit_forms.values())
+        })
+
```
Comment (hodgestar) [inline go/apps/jsbox/view_definition.py:42]: Would it be simpler to not override `get` and just access the forms in `edit_forms` directly from within the template? E.g. `edit_forms.jsbox`?

File: go/conversation/templates/conversation/dashboard.html [hunk 2]
```diff
@@ -19,6 +19,7 @@
                 <!-- TODO: We're only storing the created_at value :\ -->
                 <th>Last Edited</th>
                 <th></th>
+                <th></th>
```
Comment (hodgestar) [inline go/conversation/templates/conversation/dashboard.html:22]: Did the dashboard grow an extra column intentionally?

File: go/templates/registration/activation_complete.html [hunk 3]
```diff
@@ -1,17 +1,9 @@
-{% extends "base.html" %}
+{% extends "registration/base.html" %}
```
Comment (hodgestar) [inline go/templates/registration/activation_complete.html:1]: Did this do the wrong thing before?

**Comment by hodgestar** [comment]
I've gone through this. Left some minor questions. Let's fix the tests tomorrow morning and then have a look at it on staging with @badlard before merging to develop.

File: go/apps/jsbox/view_definition.py (hunk 1 above)
Comment (justinvdm) [inline go/apps/jsbox/view_definition.py:42]: I don't think accessing it as `edit_forms.jsbox` is possible. To make sure, I tried that before overriding `get` to find out that it doesn't work. I'd much prefer doing it that way, though, so if you have an idea to get it working that way, I'd be happy to make the change.

File: go/conversation/templates/conversation/dashboard.html (hunk 2 above)
Comment (justinvdm) [inline go/conversation/templates/conversation/dashboard.html:22]: I found out it had one `<th>` too few, so the table header border that Bootstrap 3 adds wasn't drawing properly.

Comment (hodgestar) [inline go/conversation/templates/conversation/dashboard.html:22]: Woot.

File: go/templates/registration/activation_complete.html (hunk 3 above)
Comment (justinvdm) [inline go/templates/registration/activation_complete.html:1]: Ya, previously it was redoing everything that was in base (including the logo and footer links). I assumed this was just a change that wasn't made when the UI changed.

**Comment by justinvdm** [comment]
Cool, sounds good. Happy to deploy to vumigo_solo after the comments have been resolved and Travis is happy.

File: go/templates/registration/activation_complete.html (hunk 3 above)
Comment (hodgestar) [inline go/templates/registration/activation_complete.html:1]: Tx for fixing.

**Comment by hodgestar** [comment]
:+1: once @badlard is happy with it.

**Comment by hodgestar** [comment]
:+1:. Let's land first thing tomorrow morning.

**Pull request merged**